### ISSUE TITLE
PyQt5 cannot get the correct  Unicode path  after build to exe

### ISSUE THREAD

**Issue by DionSun** [pyinstaller/pyinstaller#3696]
Dear sir,
I have the similar unicode  path issue with [Pyinstaller cannot find PyQt5 plugin directories](https://github.com/pyinstaller/pyinstaller/issues/3636).  I think it still not solve the PyQt5 Unicode path on  PyInstaller-3.4.dev0+gbbf964c6.

1. I try to print the Path in test.py
```python
import sys 
from PyQt5 import QtCore,QtGui

print("Python sys.executable:\n\t%s" % sys.executable)
print("Python sys.path:\n\t%s" % "\n\t".join(sys.path))
print('Qt5 Libraries path:\n\t%s' % \
        QtCore.QLibraryInfo.location(QtCore.QLibraryInfo.LibrariesPath))
print('Qt5 Library Executables path:\n\t%s' % \
        QtCore.QLibraryInfo.location(QtCore.QLibraryInfo.LibraryExecutablesPath))
print('Qt5 Binaries path:\n\t%s' % \
        QtCore.QLibraryInfo.location(QtCore.QLibraryInfo.BinariesPath))
print('Qt5 Data path:\n\t%s' % \
        QtCore.QLibraryInfo.location(QtCore.QLibraryInfo.DataPath))
print('Qt5 Imports path:\n\t%s' % \
        QtCore.QLibraryInfo.location(QtCore.QLibraryInfo.ImportsPath))
print('Qt5 Plugins path:\n\t%s' % \
        QtCore.QLibraryInfo.location(QtCore.QLibraryInfo.PluginsPath))
print('Qt5 Settings path:\n\t%s' % \
        QtCore.QLibraryInfo.location(QtCore.QLibraryInfo.SettingsPath))
print('Qt5 Prefix path:\n\t%s' % \
        QtCore.QLibraryInfo.location(QtCore.QLibraryInfo.PrefixPath))
```
2. And the result is:
```shell
E:\中文路径>python test.py
Python sys.executable:
        C:\Python36\python.exe
Python sys.path:
        E:\中文路径
        C:\Python36\python36.zip
        C:\Python36\DLLs
        C:\Python36\lib
        C:\Python36
        C:\Users\sunqi\AppData\Roaming\Python\Python36\site-packages
        C:\Python36\lib\site-packages
        C:\Python36\lib\site-packages\win32
        C:\Python36\lib\site-packages\win32\lib
        C:\Python36\lib\site-packages\Pythonwin
Qt5 Libraries path:
        C:/Python36/lib/site-packages/PyQt5/Qt/lib
Qt5 Library Executables path:
        C:/Python36/lib/site-packages/PyQt5/Qt/bin
Qt5 Binaries path:
        C:/Python36/lib/site-packages/PyQt5/Qt/bin
Qt5 Data path:
        C:/Python36/lib/site-packages/PyQt5/Qt
Qt5 Imports path:
        C:/Python36/lib/site-packages/PyQt5/Qt/imports
Qt5 Plugins path:
        C:/Python36/lib/site-packages/PyQt5/Qt/plugins
Qt5 Settings path:

Qt5 Prefix path:
        C:/Python36/lib/site-packages/PyQt5/Qt
```

3. Then build to exe successfully
```shell
E:\中文路径>pyinstaller test.py
68 INFO: PyInstaller: 3.4.dev0+gbbf964c6
68 INFO: Python: 3.6.5
69 INFO: Platform: Windows-10-10.0.14393-SP0
69 INFO: wrote E:\中文路径\test.spec
70 INFO: UPX is not available.
72 INFO: Extending PYTHONPATH with paths
['E:\\中文路径', 'E:\\中文路径']
72 INFO: checking Analysis
73 INFO: Building Analysis because Analysis-00.toc is non existent
73 INFO: Initializing module dependency graph...
76 INFO: Initializing module graph hooks...
77 INFO: Analyzing base_library.zip ...
3678 INFO: running Analysis Analysis-00.toc
3697 INFO: Adding Microsoft.Windows.Common-Controls to dependent assemblies of final executable
  required by c:\python36\python.exe
4561 INFO: Caching module hooks...
4566 INFO: Analyzing E:\中文路径\test.py
4632 INFO: Loading module hooks...
4632 INFO: Loading module hook "hook-encodings.py"...
4728 INFO: Loading module hook "hook-pydoc.py"...
4728 INFO: Loading module hook "hook-PyQt5.py"...
4861 WARNING: Hidden import "PyQt5.sip" not found!
4861 INFO: Loading module hook "hook-PyQt5.QtCore.py"...
4958 INFO: Loading module hook "hook-PyQt5.QtGui.py"...
5131 INFO: Loading module hook "hook-xml.py"...
5416 INFO: Looking for ctypes DLLs
5417 INFO: Analyzing run-time hooks ...
5421 INFO: Including run-time hook 'pyi_rth_qt5.py'
5426 INFO: Looking for dynamic libraries
6102 INFO: Looking for eggs
6102 INFO: Using Python library c:\python36\python36.dll
6104 INFO: Found binding redirects:
[]
6108 INFO: Warnings written to E:\中文路径\build\test\warn-test.txt
6146 INFO: Graph cross-reference written to E:\中文路径\build\test\xref-test.html
6153 INFO: checking PYZ
6154 INFO: Building PYZ because PYZ-00.toc is non existent
6154 INFO: Building PYZ (ZlibArchive) E:\中文路径\build\test\PYZ-00.pyz
6653 INFO: Building PYZ (ZlibArchive) E:\中文路径\build\test\PYZ-00.pyz completed successfully.
6659 INFO: checking PKG
6659 INFO: Building PKG because PKG-00.toc is non existent
6660 INFO: Building PKG (CArchive) PKG-00.pkg
6677 INFO: Building PKG (CArchive) PKG-00.pkg completed successfully.
6678 INFO: Bootloader c:\python36\lib\site-packages\PyInstaller\bootloader\Windows-64bit\run.exe
6678 INFO: checking EXE
6680 INFO: Building EXE because EXE-00.toc is non existent
6680 INFO: Building EXE from EXE-00.toc
6682 INFO: Appending archive to EXE E:\中文路径\build\test\test.exe
6689 INFO: Building EXE from EXE-00.toc completed successfully.
6692 INFO: checking COLLECT
6693 INFO: Building COLLECT because COLLECT-00.toc is non existent
6694 INFO: Building COLLECT COLLECT-00.toc
7159 INFO: Building COLLECT COLLECT-00.toc completed successfully.
```
 4. test.exe can not print the right PyQt5 unicode path.
```shell
E:\中文路径\dist\test>E:\中文路径\dist\test\test.exe
Python sys.executable:
        E:\中文路径\dist\test\test.exe
Python sys.path:
        E:\中文路径\dist\test\base_library.zip
        E:\中文路径\dist\test
Qt5 Libraries path:
        E:/ÖÐÎÄÂ·¾¶/dist/test/PyQt5/Qt/lib
Qt5 Library Executables path:
        E:/ÖÐÎÄÂ·¾¶/dist/test/PyQt5/Qt/bin
Qt5 Binaries path:
        E:/ÖÐÎÄÂ·¾¶/dist/test/PyQt5/Qt/bin
Qt5 Data path:
        E:/ÖÐÎÄÂ·¾¶/dist/test/PyQt5/Qt
Qt5 Imports path:
        E:/ÖÐÎÄÂ·¾¶/dist/test/PyQt5/Qt/imports
Qt5 Plugins path:
        E:/ÖÐÎÄÂ·¾¶/dist/test/PyQt5/Qt/plugins
Qt5 Settings path:

Qt5 Prefix path:
        E:/ÖÐÎÄÂ·¾¶/dist/test/PyQt5/Qt

E:\中文路径\dist\test>
```
5. There is not issue with ASCII Path  on the same test.exe
```shell
E:\test>test.exe
Python sys.executable:
        E:\test\test.exe
Python sys.path:
        E:\test\base_library.zip
        E:\test
Qt5 Libraries path:
        E:/test/PyQt5/Qt/lib
Qt5 Library Executables path:
        E:/test/PyQt5/Qt/bin
Qt5 Binaries path:
        E:/test/PyQt5/Qt/bin
Qt5 Data path:
        E:/test/PyQt5/Qt
Qt5 Imports path:
        E:/test/PyQt5/Qt/imports
Qt5 Plugins path:
        E:/test/PyQt5/Qt/plugins
Qt5 Settings path:

Qt5 Prefix path:
        E:/test/PyQt5/Qt

E:\test>
```

Thank you.

**Comment by bjones1** [comment]
This was reported to the [PyQt mailing list](https://www.riverbankcomputing.com/pipermail/pyqt/2018-August/040773.html).

**Comment by xuhcc** [comment]
Hi @DionSun 
Did you found a workaround for this problem?

I tried to set "short" windows path to plugin directory and it worked. But my project also uses qtwebengine for which I couldn't find a way to change paths to resources and locales.

```
import PyQt5
import win32api   # from pywin32 package
qt_path = os.path.join(os.path.dirname(PyQt5.__file__), 'Qt')
qt_path_short = win32api.GetShortPathName(qt_path)
os.environ['QT_PLUGIN_PATH'] = os.path.join(qt_path_short, 'plugins')
os.environ['QTWEBENGINEPROCESS_PATH'] = os.path.join(qt_path_short, 'bin')
```

Log:

```
Qt WebEngine ICU data not found at C:/Users/???/AppData/Local/Programs/GP/VideoPrice/PyQt5/Qt/resources. Trying parent directory...
Qt WebEngine ICU data not found at C:/Users/???/AppData/Local/Programs/GP/VideoPrice/PyQt5/Qt. Trying application directory...
Qt WebEngine ICU data not found at C:/Users/???/AppData/Local/Programs/GP/VideoPrice. Trying fallback directory... The application MAY NOT work.
Installed Qt WebEngine locales directory not found at location C:/Users/???/AppData/Local/Programs/GP/VideoPrice/PyQt5/Qt/translations\qtwebengine_locales. Trying application directory...
Qt WebEngine locales directory not found at location C:/Users/???/AppData/Local/Programs/GP/VideoPrice\qtwebengine_locales. Trying fallback directory... Translations MAY NOT not be correct.
```

**Comment by BoboTiG** [comment]
FTR the issue is now fixed with PyQt 5.13.0. I looked at the [PyQt changelog](https://www.riverbankcomputing.com/static/Downloads/PyQt5/5.13.0/ChangeLog) and I guess the fix is contained in one of those 3 commits:

> 2019-06-25  Phil Thompson  <phil@riverbankcomputing.com>
> 
> 	* lib/configure.py, qpy/QtCore/qpycore_api.h,
> 	qpy/QtCore/qpycore_post_init.cpp.in, qpy/QtCore/qpycore_qt_conf.cpp:
> 	The support for qt.conf is now enabled at run time rather than
> 	compile time.
> 	[59568b1d96e6] <5.12-maint>
> 
> 2019-06-24  Phil Thompson  <phil@riverbankcomputing.com>
> 
> 	* PyQt5.msp:
> 	QObject.__getattr__() now accepts a utf-8 encoded attribute name.
> 	[8d83d3192363] <5.12-maint>
> 
> 2018-09-29  Phil Thompson  <phil@riverbankcomputing.com>
> 
> 	* PyQt5.msp:
> 	Fixed QByteArrat.at() and __getitem__() to return a byte with Python
> 	v3.
> 	[8dd77fcfeebb] <5.11-maint>
